API doc generation fails with doxygen >= 1.10.0

### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Build passes, doxygen generates API doc without error.

### Current behavior

Build fails with doxygen >= 1.10.0.

The doxygen 1.10.0 [changelog](https://www.doxygen.nl/manual/changelog.html) mentions:

> Submit warning when the ingroup command is used with an non-existing group name [[view](https://github.com/doxygen/doxygen/commit/e075f1bf636ff1959dcaf0e99b9c09dc13d8008f)] and [[view](https://github.com/doxygen/doxygen/commit/9777be43fd7f2809fefcb002706977ba07c9b15c)]

and Fast-DDS's doxygen config [contains](https://github.com/eProsima/Fast-DDS/blob/v2.13.1/doxyfile.in#L839):

> WARN_AS_ERROR = YES

An inconsistency in the documentation triggers the new doxygen warning, which is being treated as error:

> `/build/source/include/fastrtps/types/AnnotationParameterValue.h:64: error: Found non-existing group 'ANNOTATIONPARAMETERVALUE' for the command '@ingroup', ignoring command (warning treated as error, aborting now)`

### Steps to reproduce

 * Get doxygen >= 1.10.0
 * Get Fast-DDS v2.13.1
 * Run cmake with `-DBUILD_DOCUMENTATION=ON`
 * Try to build FastDDS

### Fast DDS version/commit

[v2.13.1](https://github.com/eProsima/Fast-DDS/tree/v2.13.1)

https://github.com/eProsima/Fast-DDS/blob/v2.13.1/doxyfile.in#L839

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Happens on any platform that uses doxygen >= 1.10.0.

See also https://github.com/eProsima/Fast-CDR/issues/193.

### XML configuration file

_No response_

### Relevant log output

_No response_

### Network traffic capture

_No response_

--- a/include/fastrtps/config/doxygen_modules.h
+++ b/include/fastrtps/config/doxygen_modules.h
@@ -80,6 +80,16 @@
  * This module contains all classes and methods associated with RTPSReader and its specifications, as well as other necessary classes.
  */
 
+/** @defgroup TYPES_MODULE Contains the builtin generated types
+ * @namespace eprosima::fastrtps::types
+ * @ingroup FASTRTPS_MODULE
+ */
+
+/** @defgroup XMLPARSER_MODULE Contains all the modules related with the XMLParser
+ * @namespace eprosima::fastrtps::xmlparser
+ * @ingroup FASTRTPS_MODULE
+ */
+
 #ifndef DOXYGEN_SHOULD_SKIP_THIS_PUBLIC
 
 /** @defgroup PARAMETER_MODULE Qos Module

--- a/include/fastrtps/types/AnnotationParameterValue.h
+++ b/include/fastrtps/types/AnnotationParameterValue.h
@@ -59,7 +59,7 @@ namespace types {
 
 /*!
  * @brief This class represents the structure ExtendedAnnotationParameterValue defined by the user in the IDL file.
- * @ingroup ANNOTATIONPARAMETERVALUE
+ * @ingroup TYPES_MODULE
  */
 class ExtendedAnnotationParameterValue
 {
@@ -110,7 +110,8 @@ public:
      * @param current_alignment Buffer alignment.
      * @return Serialized size.
      */
-    FASTDDS_SER_METHOD_DEPRECATED(3, "eprosima::fastrtps::types::ExtendedAnnotationParameterValue::getCdrSerializedSize()",
+    FASTDDS_SER_METHOD_DEPRECATED(3,
+            "eprosima::fastrtps::types::ExtendedAnnotationParameterValue::getCdrSerializedSize()",
             "In favor of version using eprosima::fastcdr::calculate_serialized_size.")
     RTPS_DllAPI static size_t getCdrSerializedSize(
             const ExtendedAnnotationParameterValue& data,
@@ -166,7 +167,7 @@ private:
 };
 /*!
  * @brief This class represents the union AnnotationParameterValue defined by the user in the IDL file.
- * @ingroup ANNOTATIONPARAMETERVALUE
+ * @ingroup TYPES_MODULE
  */
 class AnnotationParameterValue
 {
@@ -824,7 +825,7 @@ private:
 
 /*!
  * @brief This class represents the structure AppliedAnnotationParameter defined by the user in the IDL file.
- * @ingroup ANNOTATIONPARAMETERVALUE
+ * @ingroup TYPES_MODULE
  */
 class AppliedAnnotationParameter
 {
@@ -1393,7 +1394,8 @@ public:
     }
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
-    FASTDDS_SER_METHOD_DEPRECATED(3, "eprosima::fastrtps::types::AppliedBuiltinMemberAnnotations::getCdrSerializedSize()",
+    FASTDDS_SER_METHOD_DEPRECATED(3,
+            "eprosima::fastrtps::types::AppliedBuiltinMemberAnnotations::getCdrSerializedSize()",
             "In favor of version using eprosima::fastcdr::calculate_serialized_size.")
     RTPS_DllAPI static size_t getCdrSerializedSize(
             const AppliedBuiltinMemberAnnotations& data,

--- a/include/fastrtps/types/TypeIdentifierTypes.h
+++ b/include/fastrtps/types/TypeIdentifierTypes.h
@@ -54,7 +54,7 @@ class StringLTypeDefn;
 
 /*!
  * @brief This class represents the structure StringSTypeDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class StringSTypeDefn final
 {
@@ -173,7 +173,7 @@ private:
 };
 /*!
  * @brief This class represents the structure StringLTypeDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class StringLTypeDefn final
 {
@@ -292,7 +292,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainCollectionHeader defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainCollectionHeader final
 {
@@ -436,7 +436,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainSequenceSElemDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainSequenceSElemDefn final
 {
@@ -616,7 +616,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainSequenceLElemDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainSequenceLElemDefn final
 {
@@ -796,7 +796,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainArraySElemDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainArraySElemDefn final
 {
@@ -986,7 +986,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainArrayLElemDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainArrayLElemDefn final
 {
@@ -1176,7 +1176,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainMapSTypeDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainMapSTypeDefn final
 {
@@ -1411,7 +1411,7 @@ private:
 };
 /*!
  * @brief This class represents the structure PlainMapLTypeDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class PlainMapLTypeDefn final
 {
@@ -1646,7 +1646,7 @@ private:
 };
 /*!
  * @brief This class represents the structure StronglyConnectedComponentId defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class StronglyConnectedComponentId final
 {
@@ -1829,7 +1829,7 @@ private:
 };
 /*!
  * @brief This class represents the structure ExtendedTypeDefn defined by the user in the IDL file.
- * @ingroup TYPEIDENTIFIERTYPES
+ * @ingroup TYPES_MODULE
  */
 class ExtendedTypeDefn final
 {

--- a/include/fastrtps/types/TypeObjectHashId.h
+++ b/include/fastrtps/types/TypeObjectHashId.h
@@ -46,7 +46,7 @@ typedef octet EquivalenceHash[14];
 
 /*!
  * @brief This class represents the union TypeObjectHashId defined by the user in the IDL file.
- * @ingroup TYPEOBJECTHASHID
+ * @ingroup TYPES_MODULE
  */
 class TypeObjectHashId
 {

--- a/include/fastrtps/xmlparser/XMLEndpointParser.h
+++ b/include/fastrtps/xmlparser/XMLEndpointParser.h
@@ -75,7 +75,7 @@ public:
 
 /**
  * Class XMLEndpointParser used to parse the XML file that contains information about remote endpoints.
- * @ingroup DISCVOERYMODULE
+ * @ingroup DISCOVERY_MODULE
  */
 class XMLEndpointParser : XMLParser
 {

--- a/include/fastrtps/xmlparser/XMLProfileManager.h
+++ b/include/fastrtps/xmlparser/XMLProfileManager.h
@@ -52,7 +52,7 @@ using xmlfile_map_iterator_t = xmlfiles_map_t::iterator;
 
 /**
  * Class XMLProfileManager, used to make available profiles from XML file.
- * @ingroup XMLPROFILEMANAGER_MODULE
+ * @ingroup XMLPARSER_MODULE
  */
 class XMLProfileManager
 {

